From 2a29824f84d08a9fa71094c0ab72ec2b5f3fbc79 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Mon, 1 Jun 2015 16:11:58 -0400
Subject: [PATCH] Work around LocateDevicePath() not grokking PcieRoot()
 devices properly.

Basically: PNP0A03 always has to work, so everybody always uses it, so
nobody tests PNP0A08 working, so it doesn't.

So you have to lie to the machine about which hardware you want it to
find things on, because if you tell it the truth, it'll lie to you.

The truth is a lie.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/linux.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/linux.c b/src/linux.c
index a812b0c..1231ecf 100644
--- a/src/linux.c
+++ b/src/linux.c
@@ -490,6 +490,14 @@ make_pci_path(uint8_t *buf, ssize_t size, char *pathstr, ssize_t *pathoff)
 		return -1;
 	acpi_hid = EFIDP_EFI_PNP_ID(tmp16);
 
+	/* Apparently basically nothing can look up a PcieRoot() node,
+	 * because they just check _CID.  So since _CID for the root pretty
+	 * much always has to be PNP0A03 anyway, just use that no matter
+	 * what.
+	 */
+	if (acpi_hid == EFIDP_ACPI_PCIE_ROOT_HID)
+		acpi_hid = EFIDP_ACPI_PCI_ROOT_HID;
+
 	errno = 0;
 	fbuf = NULL;
 	int use_uid_str = 0;
-- 
2.4.2

